express-notes-api/index.js

```diff
@@ -0,0 +1,135 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const notes = require('./data.json');
+
+app.get('/api/notes', function (req, res) {
+  const outputArray = [];
+  const notesData = notes.notes;
+  for (const property in notesData) {
+    outputArray.push(notesData[property]);
+  }
+  res.json(outputArray);
+});
+
+app.use(express.json());
+
+app.get('/api/notes/:id', function (req, res) {
+  const currentId = req.params.id;
+  const notesData = notes.notes;
+  let foundId = false;
+  const integerId = parseInt(currentId);
+
+  if (integerId < 0 || isNaN(integerId)) {
+    res.status(400).send('"error": "id must be a positive integer"');
+  } else {
+    for (const property in notesData) {
+      if (property === currentId) {
+        foundId = true;
+        res.json(notesData[property]);
+        fs.writeFile('data.json');
+        break;
+      }
+    }
+    if (foundId === false) {
+      res.status(404).send(`"error": "cannot find note with id ${currentId}"`);
```

Comment:
We want to use `.json()` over `.send()`. Also make sure the data you are sending is an object.

Reply:
So I should format the error message as a json object?